pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1757,60 +1702,23 @@ func (s *AWSCloud) updateInstanceSecurityGroupsForLoadBalancer(lb *elb.LoadBalan
 		return fmt.Errorf("Could not determine security group for load balancer: %s", orEmpty(lb.LoadBalancerName))
 	}
 
-	// Get the actual list of groups that allow ingress from the load-balancer
+	// Gather all security groups
 	describeRequest := &ec2.DescribeSecurityGroupsInput{}
-	filters := []*ec2.Filter{}
-	filters = append(filters, newEc2Filter("ip-permission.group-id", loadBalancerSecurityGroupId))
-	describeRequest.Filters = s.addFilters(filters)
 	actualGroups, err := s.ec2.DescribeSecurityGroups(describeRequest)
 	if err != nil {
-		return fmt.Errorf("error querying security groups: %v", err)
-	}
-
-	// Open the firewall from the load balancer to the instance
-	// We don't actually have a trivial way to know in advance which security group the instance is in
-	// (it is probably the minion security group, but we don't easily have that).
-	// However, we _do_ have the list of security groups on the instance records.
-
-	// Map containing the changes we want to make; true to add, false to remove
-	instanceSecurityGroupIds := map[string]bool{}
-
-	// Scan instances for groups we want open
-	for _, instance := range allInstances {
-		securityGroupId := findSecurityGroupForInstance(instance)
-		if isNilOrEmpty(securityGroupId) {
-			glog.Warning("Ignoring instance without security group: ", orEmpty(instance.InstanceId))
-			continue
-		}
-
-		instanceSecurityGroupIds[*securityGroupId] = true
+		glog.Error("Error querying security groups: ", err)
+		return err
```

Comment:
Let's keep `return fmt.Errorf("error querying security groups: %v", err)`  My intention is to wrap an error when it is first returned from a call to non-k8s code (e.g. an aws-sdk-go call, or a go runtime call).  And then we don't have to wrap errors from calls to our own functions - we presume they return something sensible - unless we want to add some extra contextual information.